src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -74,26 +74,6 @@ public override async Task<SelectionResult> GetValidSelectionAsync(CancellationT
                 }
             }
 
-            // Warn if local functions are in selection since data flow analysis
-            // cannot correctly analyze them
-            // https://github.com/dotnet/roslyn/issues/14214
```

Comment:
so... is this warning truly not needed anymore?  afaict, this is an orthogonal change to primary goal of this PR.  i.e. these are two separate concerns:

1. i want to be able to extract a set of code into a local function instead of a sibling method.
2. we currently warn if the selection contains a local function.

Why is '2' going away?

Reply:
I talked to Andy from the compilers team who originally added the error message. He said that this should be OK to remove since the warning was originally added to address [this](https://github.com/dotnet/roslyn/issues/14214) issue which has now been resolved.

If the selection contains a local function call and we're doing a regular extract method, I changed the code to disallow the extract method option from appearing if the selection does not also include the local function declaration, since otherwise the extracted code will always be invalid. Thus, I don't think the warning is necessary anymore.

Reply:
that makes sense.  as long as you're testing for thse cases (.e. the local-function selected or not) i'm ok with this :)

Reply:
There were some existing ones, but just added a few more with the latest commit + caught a minor bug.